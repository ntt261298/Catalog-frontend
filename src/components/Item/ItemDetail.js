import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { selectItem } from 'utils/selector';
import { getItem } from 'actions/item';
import { errMessage } from 'utils/messages';


export class ItemDetail extends Component {
  state = {
    categoryID: '',
    itemID: '',
  }

  componentDidMount() {
    const { categoryID, itemID } = this.props.params;
    this.props.getItem(categoryID, itemID)
      .catch(err => errMessage(err));
  }

  render() {
    const { item } = this.props;
    return (
      <div className="main">
        <h2>Item Detail</h2>
        <ul>
          { item.map(({ id, title, description }) => (
            <li key={id} className="user-item">
              <h3>{ title }</h3>
              <p>{ description }</p>
            </li>
          )) }
        </ul>
      </div>
    );
  }
}

ItemDetail.propTypes = {
  getItem: PropTypes.func.isRequired,
  item: PropTypes.array.isRequired,
  params: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  item: selectItem(state),
});

export default connect(mapStateToProps, {
  getItem,
})(ItemDetail);
