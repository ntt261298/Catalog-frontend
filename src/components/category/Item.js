import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import removeSpace from '../../helper/removeSpace';
import getItems from '../../actions/item';

const activeStyle = {
  background: '#444',
  color: '#fff',
};

const itemsPerPage = 1;

export class Item extends Component {
    state = {
      page: 1,
    }

    componentDidMount() {
      this.props.getCategoryItems();
    }

    setPage(page) {
      this.setState({
        page,
      });
    }

    createPage(totalPages) {
      const { page } = this.state;
      const numeric = [];
      for (let i = 1; i <= totalPages; i++) {
        numeric.push(
          <span
            onClick={this.setPage.bind(this, i)}
            style={page === i ? activeStyle : null}
          >
            { i }
          </span>,
        );
      }
      return numeric;
    }

    decreasePage() {
      const prevState = this.state.page;
      if (prevState === 1) return;
      this.setState({
        page: prevState - 1,
      });
    }

    increasePage(totalPages) {
      const prevState = this.state.page;
      if (prevState === totalPages) return;
      this.setState({
        page: prevState + 1,
      });
    }

    render() {
      const { items } = this.props;
      const { page } = this.state;
      const totalPages = Math.ceil(items.length / itemsPerPage);
      return (
        <div className="item">
          <h2>Items</h2>
          <ul>
            {
            items.slice((page - 1) * itemsPerPage, page * itemsPerPage).map(({ id, title }) => (
              <li key={id}>
                <a href={`item/${removeSpace(title)}`}>
                  { title }
                </a>
              </li>
            ))
          }
          </ul>
          <div className="numeric">
            {' '}
            <img src="assets/images/baseline-chevron_left-24px.svg" onClick={this.decreasePage.bind(this)} alt="" />
            {' '}
            <img src="assets/images/page-first.svg" onClick={this.setPage.bind(this, 1)} alt="" />
            {' '}
            { this.createPage(totalPages) }
            {' '}
            <img src="assets/images/page-last.svg" onClick={this.setPage.bind(this, totalPages)} alt="" />
            {' '}
            <img src="assets/images/baseline-chevron_right-24px.svg" onClick={this.increasePage.bind(this, totalPages)} alt="" />
            {' '}
          </div>
        </div>
      );
    }
}

Item.propTypes = {
  getCategoryItems: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  items: state.category.items,
});

export default connect(mapStateToProps, { getItems })(Item);
