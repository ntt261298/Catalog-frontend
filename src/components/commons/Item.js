import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import { Card, Button } from 'react-bootstrap';
import { getItems, getCategoryItems } from '../../actions/item';
import { selectCategoryItems, selectAllItems } from '../../utils/selector';
import { errMessage } from '../../utils/messages';


export class Item extends Component {
  componentDidMount() {
    switch (this.props.type) {
      case 'home': {
        this.props.getItems()
          .catch(err => errMessage(err));
        return;
      }
      case 'category': {
        this.props.getCategoryItems(this.props.id)
          .catch(err => errMessage(err));
        break;
      }
      default:
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.id !== this.props.id) {
      this.props.getCategoryItems(this.props.id)
        .catch(err => errMessage(err));
    }
  }

  render() {
    const { type, categoryItems, allItems } = this.props;
    return (
      <div className="item">
        { type === 'home' ? (
          <Fragment>
            <h2>All Items</h2>
            <ul className="item-list">
              {allItems.map(({ id, title, category_id }) => (
                <Card key={id} className="item-card">
                  <Card.Body>
                    <Card.Title style={{ borderBottom: '1px solid #f7f7f7' }}>{ title }</Card.Title>
                    <Link to={`/category/${category_id}/item/${id}`}>
                      <Button variant="primary">
                          View Detail
                      </Button>
                    </Link>
                  </Card.Body>
                </Card>
              ))}
            </ul>
          </Fragment>
        ) : (
          <Fragment>
            <h2>Category's Items</h2>
            <ul className="item-list">
              {categoryItems.map(({ id, title, category_id }) => (
                <Card key={id} className="item-card">
                  <Card.Body>
                    <Card.Title style={{ borderBottom: '1px solid #f7f7f7' }}>{ title }</Card.Title>
                    <Link to={`/category/${category_id}/item/${id}`}>
                      <Button variant="primary">
                        View Detail
                      </Button>
                    </Link>
                  </Card.Body>
                </Card>
              ))}
            </ul>
          </Fragment>
        )}
      </div>
    );
  }
}

Item.propTypes = {
  getItems: PropTypes.func.isRequired,
  getCategoryItems: PropTypes.func.isRequired,
  categoryItems: PropTypes.array.isRequired,
  allItems: PropTypes.array.isRequired,
  type: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  categoryItems: selectCategoryItems(state),
  allItems: selectAllItems(state),
});

export default connect(mapStateToProps, { getItems, getCategoryItems })(Item);
