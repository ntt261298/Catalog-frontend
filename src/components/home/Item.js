import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';

class Item extends Component {
    render() {
        return (
            <div className="Item">
                <h2>All Items</h2>
                <ul>
                    <li><a href=""></a>Sport</li>
                    <li><a href=""></a>Book</li>
                    <li><a href=""></a>Furniture</li>
                    <li><a href=""></a>Cloth</li>
                    <li><a href=""></a>Technology</li>
                    <li><a href=""></a>Food</li>
                </ul>
                <div class="numeric">
                    <img src="assets/images/baseline-chevron_left-24px.svg" alt=""/>
                    <img src="assets/images/page-first.svg" alt=""/>
                    <span>1</span>
                    <span>2</span>
                    <span>3</span>
                    <span>4</span>
                    <span>5</span>
                    <img src="assets/images/page-last.svg" alt=""/>
                    <img src="assets/images/baseline-chevron_right-24px.svg" alt=""/>
                </div>
            </div>
        )
    }
}

Item.propTypes = {
    // toggleLogin: PropTypes.func.isRequired,
    // userLogout: PropTypes.func.isRequired,
    // getSearchResults: PropTypes.func.isRequired
  }
  
  const mapStateToProps = state => ({

  })

export default connect(null, null)(Item);
