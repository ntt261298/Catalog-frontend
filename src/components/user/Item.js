import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { getUserItems } from '../../actions/item';
import lastPage from '../../assets/images/page-last.svg';
import firstPage from '../../assets/images/page-first.svg';
import left from '../../assets/images/left.svg';
import right from '../../assets/images/right.svg';


const activeStyle = {
  background: '#444',
  color: '#fff',
};

const itemsPerPage = 1;

export class Item extends Component {
    state = {
      page: 1,
    }

    componentDidMount() {
      this.props.getUserItems(this.props.token);
    }

    setPage(page) {
      this.setState({
        page,
      });
    }

    createPage(totalPages) {
      const { page } = this.state;
      const numeric = [];
      for (let i = 1; i <= totalPages; i++) {
        numeric.push(
          <span
            onClick={this.setPage.bind(this, i)}
            key={i}
            style={page === i ? activeStyle : null}
          >
            { i }
          </span>,
        );
      }
      return numeric;
    }

    decreasePage() {
      const prevState = this.state.page;
      if (prevState === 1) return;
      this.setState({
        page: prevState - 1,
      });
    }

    increasePage(totalPages) {
      const prevState = this.state.page;
      if (prevState === totalPages) return;
      this.setState({
        page: prevState + 1,
      });
    }

    render() {
      const { items } = this.props;
      const { page } = this.state;
      const totalPages = Math.ceil(items.length / itemsPerPage);
      return (
        <div className="item">
          <h2>User's Items</h2>
          <ul>
            {
            items.slice((page - 1) * itemsPerPage, page * itemsPerPage).map(
              ({ id, title, category_id }) => (
                <li key={id}>
                  <a href={`category/${category_id}/item/${id}`}>
                    { title }
                  </a>
                </li>
              ),
            )
          }
          </ul>
          <div className="numeric">
            {' '}
            <img src={left} onClick={this.decreasePage.bind(this)} alt="" />
            {' '}
            <img src={firstPage} onClick={this.setPage.bind(this, 1)} alt="" />
            {' '}
            { this.createPage(totalPages) }
            {' '}
            <img src={lastPage} onClick={this.setPage.bind(this, totalPages)} alt="" />
            {' '}
            <img src={right} onClick={this.increasePage.bind(this, totalPages)} alt="" />
            {' '}
          </div>
        </div>
      );
    }
}

Item.propTypes = {
  getUserItems: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
  token: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  items: state.item.items,
  token: state.user.accessToken,
});

export default connect(mapStateToProps, { getUserItems })(Item);
