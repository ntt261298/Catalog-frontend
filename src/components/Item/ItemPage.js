import React from 'react';
import { connect } from 'react-redux';
import 'style/home.css';
import { PropTypes } from 'prop-types';
import ItemDetail from 'components/Item/ItemDetail';
import Header from 'components/Layout/Header';

export const ItemPage = props => (
  <div>
    <Header />
    <div className="item-main">
      <ItemDetail params={props.match.params} />
    </div>
  </div>
);

ItemPage.propTypes = {
  match: PropTypes.object.isRequired,
};

export default connect(null, null)(ItemPage);
