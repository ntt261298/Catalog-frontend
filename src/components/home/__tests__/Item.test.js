import React from 'react';
import { shallow } from '../../../enzyme';
import Item from '../Item';

describe('Item tests', () => {
  let wrapper;
  let props;
  const setup = () => {
    wrapper = shallow(
      <Item {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      ...props,
      items: [{ id: 1, title: 'Title 1', desciption: 'One' },
        { id: 2, title: 'Title 2', desciption: 'Two' },
        { id: 3, title: 'Title 3', desciption: 'Three' }],
      getItems: () => props.items,
    };
  });

  it('renders list-items', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
