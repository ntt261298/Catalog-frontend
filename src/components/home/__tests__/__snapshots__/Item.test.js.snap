// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item tests renders list-items 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
